### PR TITLE
[MPS] Flatten 5D tensors to 4D in batch_norm for performance

### LINKED ISSUE #180334: [MPS] MPSGraph normalization is significantly slower for rank-5 tensors
MPSGraph's normalization ops (`normalizationWithTensor:`, `normalizationGradientWithIncomingGradientTensor:`) are significantly slower for rank-5 (BatchNorm3d) inputs compared to rank-4 (BatchNorm2d) with the same number of elements.

Since BatchNorm reduces over all dimensions except the channel dim, spatial dimensions can be safely merged: `[N, C, D, H, W]` → `[N, C, D*H, W]`.

### Benchmark (M4 Pro, PyTorch main @ 0b9170d)

```
native_batch_norm forward (shape: [4, 64, 64, 64, 64]):
  5D [N,C,D,H,W]:     4.5 ms
  4D [N,C,D*H,W]:     4.5 ms  ← after flatten

native_batch_norm_backward:
  5D [N,C,D,H,W]:     6.7 ms
  4D [N,C,D*H,W]:     6.8 ms  ← after flatten

nn.BatchNorm3d fwd+bwd (before fix):  8.7 ms
nn.BatchNorm2d fwd+bwd (equivalent):  3.5 ms  ← 2.4x faster
```

### Reproducer

```python
import time
import torch
import torch.nn as nn

assert torch.backends.mps.is_available()
device = torch.device("mps")

N, C, D, H, W = 4, 64, 64, 64, 64
steps = 20

def bench(label, fn):
    for _ in range(5):
        fn()
    torch.mps.synchronize()
    t0 = time.perf_counter()
    for _ in range(steps):
        fn()
    torch.mps.synchronize()
    ms = (time.perf_counter() - t0) / steps * 1000
    print(f"{label}: {ms:.1f} ms")
    return ms

bn3d = nn.BatchNorm3d(C).to(device).train()
bn2d = nn.BatchNorm2d(C).to(device).train()
bn2d.load_state_dict(bn3d.state_dict())

x5 = torch.randn(N, C, D, H, W, device=device, requires_grad=True)
x4 = torch.randn(N, C, D*H, W, device=device, requires_grad=True)

t5 = bench("BatchNorm3d (5D) fwd+bwd", lambda: bn3d(x5).sum().backward())
t4 = bench("BatchNorm2d (4D) fwd+bwd", lambda: bn2d(x4).sum().backward())
print(f"Ratio: {t5/t4:.1f}x")
```

cc @jerryzh168 @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk

### PR DESCRIPTION
**Pull request by nuri-yoo**
## Summary

MPSGraph normalization ops (`normalizationWithTensor:`, `normalizationGradientWithIncomingGradientTensor:`) are significantly slower for rank-5 tensors (BatchNorm3d) compared to rank-4 (BatchNorm2d) with the same number of elements.

Since BatchNorm reduces over all dimensions except the channel dim, spatial dimensions can be safely merged: `[N, C, D, H, W]` → `[N, C, D*H, W]`. This PR adds a 5D→4D reshape at the top of `batch_norm_mps_out` and `batch_norm_backward_mps`, then recurses into the existing 4D path.

## Benchmark (M4 Pro, shape [4, 64, 64, 64, 64])

| | Before | After |
|---|---|---|
| nn.BatchNorm3d fwd+bwd | 8.7 ms | 3.5 ms (**2.4x**) |
| native_batch_norm (forward) | 4.5 ms | 4.5 ms (parity with 4D) |
| native_batch_norm_backward | 6.7 ms | 6.8 ms (parity with 4D) |

## Test plan

All 5 existing MPS batch_norm tests pass:

```
test/test_mps.py::TestMPS::test_batch_norm PASSED
test/test_mps.py::TestMPS::test_batch_norm_backward PASSED
test/test_mps.py::TestMPS::test_batch_norm_backward_weight_bias_gradients PASSED
test/test_mps.py::TestMPS::test_batch_norm_mixed_dtype PASSED
test/test_mps.py::TestMPS::test_batch_norm_slices PASSED
```

Correctness: max diff between 5D and equivalent 4D is 2.38e-07 (float32 machine epsilon).

Fixes #180334

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by nuri-yoo** [comment]
The `module: inductor` and `release notes: inductor (aoti)` labels were picked up from a stale branch before force-push. This PR only touches MPS batch_norm — could a maintainer remove those labels?

**Comment by malfet** [comment]
@claude review this one, more specifically confirm that mentioned tests indeed cover batch_norm 3d

**Review by malfet — APPROVED** [review]

**Comment by nuri-yoo** [comment]
@malfet Could you approve the pending workflow runs (BC Lint and pull)? They are stuck on `action_required` since I am a first-time contributor. All other checks have passed. Could you also run `@pytorchbot merge` after they pass?
